ledger-core/inc/core/operation/OperationDatabaseHelper.hpp

```diff
@@ -0,0 +1,200 @@
+ * SOFTWARE.
+ *
+ */
+
+#pragma once
+
+#include <string>
+
+#include <soci.h>
+
+#include <core/api/OperationType.hpp>
+#include <core/operation/Operation.hpp>
+
+namespace ledger {
+    namespace core {
+
+        template <typename Derived>
+        class OperationDatabaseHelper {
+        public:
+            static bool putOperation(soci::session& sql, const Operation& operation);
+            
+            static std::string createUid(const std::string& accountUid,
+                                            const std::string& txId,
+                                            const api::OperationType type);
+            static void queryOperations(soci::session& sql, int32_t from, int32_t to,
+                                        bool complete, bool excludeDropped, std::vector<Operation>& out);
+
+            static std::size_t queryOperations(soci::session &sql,
+                                                const std::string &accountUid,
+                                                std::vector<Operation>& out,
+                                                std::function<bool (const std::string& address)> filter);
+        };
+
+        namespace impl {
+
+            // Declares a type trait to check if templated type has a static method
+            // called `updateCurrencyOperation`.
+            template <typename T, typename = void>
+            struct has_update;
+
+            // Uses SFINAE to determine if templated type has the 
+            // `updateCurrencyOperation` method.
+            // If the substitution succeed, `has_update` is evaluated as a
+            // `true_type`
+            template <typename T>
+            struct has_update<T, decltype(T::updateCurrencyOperation)> : std::true_type
+            {};
+
+            // Else the `has_update` is evaluated as a `false_type`
+            template <typename T>
+            struct has_update<T, void> : std::false_type
+            {};
+
+            // Syntaxic sugar here to avoid to write the `has_update<T>::value`
+            template <typename T>
+            constexpr auto has_update_v = has_update<T>::value;
+        }
+
+        template <typename Derived>
+        bool OperationDatabaseHelper<Derived>::putOperation(soci::session& sql, const Operation& operation)
+        {
+            static_assert(
+                impl::has_update_v<Derived>,
+                "OperationDatabaseHelper<T> requires 'T' to define 'updateCurrencyOperation' method");
+
+            using namespace soci;
+
+            auto count = 0;
+            std::string serializedTrust;
+            // TODO: I'm honestly not sure at all about this line below - since
+            // api::TrustIndicator has none `serialize` method, we obviously need the
+            // underlying implementation. Change `api::TrustIndicator` to
+            // `TrustIndicator` in the `Operation` class is for sure a better solution 
```

Comment:
As we discussed live, it’d be better to move `TrustIndicator` in the modularization and change from `api::TrustIndicator` to `TrustIndicator` in `Operation` so that you don’t require that dynamic cast.